Extensions/xkit_patches.js

```diff
@@ -252,7 +252,43 @@ XKit.extensions.xkit_patches = new Object({
 						onload: resolve,
 						onerror: reject
 					});
-				})
+				}),
+
+				post: {
+					fetch: data => new Promise((resolve, reject) => {
+						XKit.tools.Nx_XHR({
+							method: "GET",
+							url: "https://www.tumblr.com/svc/post/fetch?" + $.param(data),
+							onload: resolve,
+							onerror: reject
```

Comment:
we shouldn't be rejecting things that aren't Error types, so we should wrap the response in a `new Error`

Reply:
this would require us to stringify the response object and then parse it in every `.catch` though. i feel like the theoretical correctness of `.catch` always having an Error isn't worth that mess

Reply:
unless you have a more specific idea of what to feed the `new Error`?

Reply:
@nightpool are you willing to compromise on this point? because if not, every other svc function will have to be rewritten to satisfy it too

Reply:
sure, although we wouldn't need to stringify anything, we'd just attach the errors object to the Error directly, and then have a human-readable error message (i.e. `errors.message` or whatever tumblr calls it)